### PR TITLE
Stereochemistry-related SWIG updates

### PR DESCRIPTION
**Pull request by ptosco**
- expose `[sg]etUseLegacyStereo()`
- In `MolToSmiles()` `doIsomericSmiles` should default to `true` as in C++ and Python
- added missing parameters to `MolToSmiles()` and `MolToMolBlock()`
- added `SmilesWriteParams` `MolToSmiles()` overload
- added and updated Java tests

### CODE REVIEW

File: Code/JavaWrappers/ROMol.i [hunk 1]
```diff
@@ -155,6 +156,18 @@ void setPreferCoordGen(bool val) {
...
+  return RDKit::Chirality::getUseLegacyStereoPerception();
+}
+void setUseLegacyStereoPerception(bool val) {
+  RDKit::Chirality::setUseLegacyStereoPerception(val);
+}
+%}
+
+bool getUseLegacyStereoPerception();
+void setUseLegacyStereoPerception(bool);
+
```
Comment (greglandrum) [inline Code/JavaWrappers/ROMol.i:170]: Let's expose the non-tetrahedral chirality flag at the same time:
```suggestion
%{
bool getUseLegacyStereoPerception() {
  return RDKit::Chirality::getUseLegacyStereoPerception();
}
void setUseLegacyStereoPerception(bool val) {
  RDKit::Chirality::setUseLegacyStereoPerception(val);
}
bool getAllowNontetrahedralChirality() {
  return RDKit::Chirality::getAllowNontetrahedralChirality();
}
void setAllowNontetrahedralChirality(bool val) {
  RDKit::Chirality::setAllowNontetrahedralChirality(val);
}
%}

bool getUseLegacyStereoPerception();
void setUseLegacyStereoPerception(bool);
bool getAllowNontetrahedralChirality();
void setAllowNontetrahedralChirality(bool);
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
One small suggestion.

Before merging, this also needs an entry in the `Backwards incompatible changes` section of `ReleaseNotes.md` and in `BackwardIncompatibleChanges.md` documenting the changes to the default behavior of `MolToSmiles`. I agree that the change is the right thing to do, but since it is guaranteed to break people's workflows, we want to be as careful as possible to document it well.

File: Code/JavaWrappers/ROMol.i (hunk 1 above)
Comment (ptosco) [reply Code/JavaWrappers/ROMol.i]: @greglandrum I have applied your suggestions, and also exposed the same functionality in MinimalLib and CFFI (with tests).

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**